storage/storage_transport.go

```diff
@@ -0,0 +1,77 @@
+package storage
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/storage"
+)
+
+var (
+	// Transport is an ImageTransport that uses a default storage.Mall or
+	// one that's it's explicitly told to use.
+	Transport MallTransport = &storageTransport{}
+	// ErrUnknownPolicyConfigurationScope is returned when
+	// ValidatePolicyConfigurationScope() fails.
+	ErrUnknownPolicyConfigurationScope = errors.New("unknown policy configuration scope")
+	// ErrInvalidReference is returned when ParseReference() is passed an
+	// empty reference.
+	ErrInvalidReference = errors.New("invalid reference")
+	idRegexp            = regexp.MustCompile("^(sha256:)?([0-9a-fA-F]{64})$")
+)
+
+// MallTransport is an ImageTransport that uses a storage.Mall, either its own
+// or one that it's told to use.
+type MallTransport interface {
+	types.ImageTransport
+	SetMall(storage.Mall) error
+}
+
+type storageTransport struct {
+	mall storage.Mall
+}
+
+func (s *storageTransport) Name() string {
+	return "oci-storage"
+}
+
+// SetMall sets the Mall object which the Transport will use.  If one is not
+// explicitly provided, the library will attempt to initialize one with default
+// settings the first time a reference needs to be parsed.
+func (s *storageTransport) SetMall(mall storage.Mall) error {
+	s.mall = mall
+	return nil
+}
+
+// ParseReference initializes the storage library, and then takes a name or an
+// ID, tries to figure out which it is, and returns it in a reference object.
+func (s *storageTransport) ParseReference(reference string) (types.ImageReference, error) {
+	if reference == "" {
+		return nil, ErrInvalidReference
+	}
+	if s.mall == nil {
+		mall, err := storage.MakeMall("", "", "", []string{})
+		if err != nil {
+			return nil, err
+		}
+		s.mall = mall
+	}
+	id := ""
+	if matches := idRegexp.FindStringSubmatch(reference); len(matches) > 1 {
+		id = matches[len(matches)-1]
+		logrus.Debugf("parsed reference %q into ID %q", reference, id)
+		reference = ""
+	} else {
+		logrus.Debugf("treating reference %q as a name", reference)
+	}
+	return newReference(s, reference, id), nil
+}
+
+func (s *storageTransport) ValidatePolicyConfigurationScope(scope string) error {
+	if scope != "" {
```

Comment:
So, this always fails, and a signature verification policy can only be defined for the whole transport.